### PR TITLE
Add retain_repo_versions filter

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #3507

### CODE REVIEW

File: pulpcore/app/viewsets/repository.py [hunk 1]
```diff
@@ -39,7 +39,10 @@ class RepositoryFilter(BaseFilterSet):
 
     class Meta:
         model = Repository
-        fields = {"name": NAME_FILTER_OPTIONS}
+        fields = {
+            "name": NAME_FILTER_OPTIONS,
+            "retain_repo_versions": ["exact", "isnull"] + DATETIME_FILTER_OPTIONS,
```
Comment (mdellweg) [inline pulpcore/app/viewsets/repository.py:44]: Hmm, is what is `DATETIME_FILTER_OPTION`? `retain_repo_version` is surely not a `datetime`. So should it be called `NUMERICAL_COMPARE_FILTER_OPTIONS`?

Comment (gerrod3) [reply pulpcore/app/viewsets/repository.py:44]: `DATETIME_FILTER_OPTIONS = ["lt", "lte", "gt", "gte", "range"]`, the filterset auto-builds the filters based on the operations and the type of field it is.

Comment (mdellweg) [reply pulpcore/app/viewsets/repository.py:44]: I guess this doing exactly what we want. My point is i would not dare to try using something called "for datetime" when the objects i filter for are integers. So i think we should rename that constant if we can. Is it in the plugin api?

Comment (gerrod3) [reply pulpcore/app/viewsets/repository.py:44]: Yes it is in the plugin api. Should I just write out the whole list instead of using a constant?

Comment (mdellweg) [reply pulpcore/app/viewsets/repository.py:44]: So we'd need to duplicate the constant to give it a proper name...
I am going to approve and leave it to your best judjment.

**Review by mdellweg — APPROVED** [review]

File: pulpcore/app/viewsets/repository.py (hunk 1 above)
Comment (gerrod3) [reply pulpcore/app/viewsets/repository.py]: I created a new constant `NULLABLE_NUMERIC_FILTER_OPTIONS` and added to the plugin API.

**Review by ipanova — APPROVED** [review]

**Pull request merged**